### ISSUE TITLE
Implement user dashboard

### ISSUE THREAD

**Issue by enjolras** [TheOnlineJudge/ojudge#19]
DashboardWidget must be completed with what could be reasonable to have in it:

- Lastest submissions.
- Current rankings.
- Coming contests.
- Recommended problems.
- Etc.

**Comment by ProBeta1** [comment]
Plus :
- User problem solving stats ( using graphs maybe )
And how the problems will be recommended ?

**Comment by enjolras** [comment]
Totally agree with the user own stats (maybe the most important of all the info in the dashboard :D ). About problems to be solved, we will have to figure some algorithm. Maybe some similar to uHunt, maybe some IA based on previous "paths" followed by others users. We have ~25 million submissions, so we will figure something. Any experts on IA and neural networks in the room?

**Comment by ProBeta1** [comment]
Can we somehow sort the problems to some kind of rating level ? It would make recommendations super easy !

**Comment by enjolras** [comment]
Probably "playing" with the totalsubmissions/accepted ratio, or accepted at first/second/... submission percent, or something around that could be a good basis for a (simple) ranking. Other, more complex, algorithms could be designed as well.

**Comment by ProBeta1** [comment]
I dont think that the first approach of AC/attempts ratio would be a good criteria  because maybe people haven't just came across that problem .. Sites like Codeforces have rated users so allotting rating to new problems is not a big deal there . But in our case we do not have any such information about the users who have submitted the solutions , so maybe we should rely on neural networks ...
